lib/core/src/irods_client_negotiation.cpp

```diff
@@ -43,39 +44,46 @@ namespace irods
 
     /// =-=-=-=-=-=-=-
     /// @brief given a property map and the target host name decide between a federated key and a local key
-    const std::string& determine_negotiation_key( const std::string& _host_name )
+    auto determine_negotiation_key(const std::string& _host_name) -> std::string
     {
-        // search the federation map for the host name
         try {
-            for ( const auto& el : irods::get_server_property<const std::vector<boost::any>>(irods::CFG_FEDERATION_KW) ) {
+            for (const auto& el : irods::get_server_property<const std::vector<boost::any>>(irods::CFG_FEDERATION_KW)) {
```

Comment:
Should the vector be a reference?

Reply:
I was wondering this myself. I didn't change it because this was the original type, but I'll give it a try and see how it goes.

Reply:
The good thing about this change is that the objects are marked as const so the compiler would've told us if we violated the constness of something (assuming we aren't doing anything weird in there lol).